### PR TITLE
[JENKINS-47500] Secret.decrypt launches ArrayIndexOutOfBoundsException when data is "{}"

### PR DESCRIPTION
**Pull request by escoem**
See [JENKINS-47500](https://issues.jenkins-ci.org/browse/JENKINS-47500).

unit test for bug added.

### Desired reviewers
@oleg-nenashev

### CODE REVIEW

**Review by daniel-beck — APPROVED** [review]

File: core/src/test/java/hudson/util/SecretUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package hudson.util;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.jvnet.hudson.test.Issue;
+
+public class SecretUtilTest {
```
Comment (daniel-beck) [inline core/src/test/java/hudson/util/SecretUtilTest.java]: SecretUtilTest?

Comment (oleg-nenashev) [reply core/src/test/java/hudson/util/SecretUtilTest.java]: Yeah, just "SecretTest" would be fine

**Review by oleg-nenashev — APPROVED** [review]
🐝 and @reviewbybees done

File: core/src/test/java/hudson/util/SecretUtilTest.java (hunk 1 above)
Comment (escoem) [reply core/src/test/java/hudson/util/SecretUtilTest.java]: @daniel-beck @oleg-nenashev I had problems with https://github.com/jenkinsci/jenkins/blob/master/core/src/test/groovy/hudson/util/SecretTest.groovy when I named it as SecretTest.java

That's why I use SecretUtilTest.java

**Comment by daniel-beck** [comment]
Would be interesting to know the use case here. Who creates secrets like this?

**Comment by stephenc** [comment]
@daniel-beck so one of the use cases for templating files is that you can lay down the file with the "secret" in plain text and then, because it will not decrypt, it will be picked up and encrypted and then when saved it will be replaced with the encrypted form.

So if you happen to have a plain text secret of `{}` then you will be SooL... now that is a really crummy short secret, but we do know that the actual secrets have to have at least some content... so we should know therefore that `{}` or `{A}` or `{pass}` are all too short to actually be a secret and then not even try to decrypt them

**Comment by daniel-beck** [comment]
@stephenc makes sense, thanks.

**Review by oleg-nenashev — APPROVED** [review]
The test code would benefit from copy-paste cleanup, but LGTM in general

**Comment by daniel-beck** [comment]
Please don't ignore the submitter checklist in the future.

**Comment by escoem** [comment]
ack @daniel-beck I won't

**Pull request merged**